ui/src/services/alertUtils.js

```diff
@@ -181,3 +182,42 @@ export const getHealthStatus = (
     return STATUS_WARNING;
   return STATUS_NONE;
 };
+
+/*
+Format the alerts from Loki.
+We need to remove the alerts with the same ID, because the same alert may be retriggered by multiple times.
+*/
+export const formatHistoryAlerts = (streamValues: StreamValue): Alert[] => {
+  const alerts = streamValues.map((value) => {
+    const alert = JSON.parse(value.values[1]);
+    return {
+      id: alert.fingerprint,
+      summary: alert.annotations.summary,
+      description: alert.annotations.description || alert.annotations.message,
+      startsAt: alert.startsAt,
+      endsAt: alert.endsAt || new Date().toISOString(),
+      severity: alert.labels.severity,
+      documentationUrl: alert.annotations.runbook_url,
+      labels: {
+        ...alert.labels,
+        parents:
+          (alert.annotations.parents && alert.annotations.parents.split(',')) ||
+          [],
+        selectors:
+          (alert.annotations.selectors &&
+            alert.annotations.selectors.split(',')) ||
+          [],
+      },
+      originalAlert: alert,
+    };
+  });
+
+  return alerts.reduce((acc, current) => {
+    const x = acc.find((item) => item.id === current.id);
+    if (!x) {
+      return acc.concat([current]);
+    } else {
+      return acc;
+    }
+  }, []);
```

Comment:
Maybe in order to simplify this we can index the alerts by their ids and then keep the values of this object.

```suggestion
  const alerts = streamValues.reduce((agg, value) => {
    const alert = JSON.parse(value.values[1]);
    return {
       ...agg,
       [alert.fingerprint]: {
          id: alert.fingerprint,
          summary: alert.annotations.summary,
          description: alert.annotations.description || alert.annotations.message,
          startsAt: alert.startsAt,
          endsAt: alert.endsAt || new Date().toISOString(),
          severity: alert.labels.severity,
          documentationUrl: alert.annotations.runbook_url,
          labels: {
            ...alert.labels,
            parents:
              (alert.annotations.parents && alert.annotations.parents.split(',')) ||
              [],
            selectors:
              (alert.annotations.selectors &&
                alert.annotations.selectors.split(',')) ||
              [],
       }
    },
      originalAlert: alert,
    };
  }, {});

  return Object.values(alerts);
```